## 📋 Quality Gate Results

| Check | Status |
|-------|--------|
| Knip | ✅ success |
| Lint | ❌ failure |
| Build | ✅ success |
| Infra Tests | ❌ skipped |
| Unit Tests | ❌ skipped |
| Component Tests | ❌ skipped |
| Perf Tests | ❌ skipped |
| Visual Tests | ❌ skipped |


### ❌ Lint Failure Details
```text

> hrm@0.28.0 lint /home/runner/work/hrm/hrm
> eslint app/ components/ constants/ context/ hooks/ lib/ services/ tests/ types/ utils/ server.ts proxy.ts --cache


/home/runner/work/hrm/hrm/tests/unit/hooks/useSpotifyAuth.test.ts
  42:37  error  Replace `·data:·mockSession,·status:·'authenticated'` with `⏎······data:·mockSession,⏎······status:·'authenticated',⏎···`    prettier/prettier
  65:37  error  Replace `·data:·mockSession,·status:·'authenticated'` with `⏎······data:·mockSession,⏎······status:·'authenticated',⏎···`    prettier/prettier
  81:37  error  Replace `·data:·mockSession,·status:·'authenticated'` with `⏎······data:·mockSession,⏎······status:·'authenticated',⏎···`    prettier/prettier
  90:37  error  Replace `·data:·validSession,·status:·'authenticated'` with `⏎······data:·validSession,⏎······status:·'authenticated',⏎···`  prettier/prettier

✖ 4 problems (4 errors, 0 warnings)
  4 errors and 0 warnings potentially fixable with the `--fix` option.

 ELIFECYCLE  Command failed with exit code 1.
```
### ❌ Infrastructure Test Failure Details
```text
Log file not found.
```
### ❌ Unit Test Failure Details
```text
Log file not found.
```
### ❌ Component Test Failure Details
```text
Log file not found.
```
### ❌ Visual Test Failure Details
```text
Log file not found.
```
### ❌ Performance Test Failure Details
```text
Log file not found.
```
⚠️ **Some checks failed.** Full logs available in [workflow artifacts](https://github.com/arii/hrm/actions/runs/21466014010).

---
> Report generated for commit: `2efbdaaefc54d6ec1aa7d7f8c87d32bd5becd13f`

--- a/hooks/useSpotifyAuth.ts
+++ b/hooks/useSpotifyAuth.ts
@@ -29,7 +29,10 @@ export const useSpotifyAuth = () => {
         persist: true,
       })
       // Automatically sign out the user to clear the invalid session
-      signOut()
+      const signOutUser = async () => {
+        await signOut()
+      }
+      signOutUser()
     }
   }, [session, addError])
 

--- a/hooks/useSpotifyWebPlayback.ts
+++ b/hooks/useSpotifyWebPlayback.ts
@@ -99,6 +99,11 @@ const useSpotifyWebPlayback = () => {
           addError('Spotify session expired. Please log in again.', {
             persist: true,
           })
+          // Note: This is a redundant sign-out trigger. The `useSpotifyAuth`
+          // hook also handles the `RefreshAccessTokenError` and initiates a
+          // sign-out. While this is a safeguard, a future refactor could
+          // streamline this to rely on a single source of truth for session
+          // validity.
           await signOut()
         }
         // For other errors, we might not need to sign out.

--- a/tests/unit/hooks/useSpotifyAuth.test.ts
+++ b/tests/unit/hooks/useSpotifyAuth.test.ts
@@ -39,7 +39,10 @@ describe('useSpotifyAuth', () => {
 
   it('should return authenticated status when session exists', () => {
     const mockSession = { accessToken: '123', user: { name: 'Test User' } }
-    mockUseSession.mockReturnValue({ data: mockSession, status: 'authenticated' })
+    mockUseSession.mockReturnValue({
+      data: mockSession,
+      status: 'authenticated',
+    })
 
     const { result } = renderHook(() => useSpotifyAuth())
 
@@ -62,7 +65,10 @@ describe('useSpotifyAuth', () => {
       error: 'RefreshAccessTokenError',
       accessToken: 'expired-token',
     }
-    mockUseSession.mockReturnValue({ data: mockSession, status: 'authenticated' })
+    mockUseSession.mockReturnValue({
+      data: mockSession,
+      status: 'authenticated',
+    })
 
     renderHook(() => useSpotifyAuth())
 
@@ -78,7 +84,10 @@ describe('useSpotifyAuth', () => {
       error: 'SomeOtherError',
       accessToken: 'valid-token',
     }
-    mockUseSession.mockReturnValue({ data: mockSession, status: 'authenticated' })
+    mockUseSession.mockReturnValue({
+      data: mockSession,
+      status: 'authenticated',
+    })
 
     const { rerender } = renderHook(() => useSpotifyAuth())
 
@@ -87,7 +96,10 @@ describe('useSpotifyAuth', () => {
 
     // Rerender with no error
     const validSession = { accessToken: 'valid-token' }
-    mockUseSession.mockReturnValue({ data: validSession, status: 'authenticated' })
+    mockUseSession.mockReturnValue({
+      data: validSession,
+      status: 'authenticated',
+    })
     rerender()
 
     expect(mockAddError).not.toHaveBeenCalled()

